test-tools/core/src/main/java/org/kie/cloud/tests/core/resources/Project.java

```diff
@@ -0,0 +1,329 @@
+            LOGGER.info("Create project {}", projectName);
+            openShift.createProjectRequest(projectName);
+            new SimpleWaiter(() -> openShift.getProject(projectName) != null)
+                    .reason("Waiting for project " + projectName + " to be created.")
+                    .timeout(TimeUnit.MINUTES, 5)
+                    .waitFor();
+        }
+        return new Project(projectName);
+    }
+
+    /**
+     * @return Project name.
+     */
+    public String getName() {
+        return projectName;
+    }
+
+    /**
+     * @return OpenShift client.
+     */
+    public OpenShift getOpenShift() {
+        return openShift;
+    }
+
+    /**
+     * @return OpenShift admin client.
+     */
+    public OpenShift getOpenShiftAdmin() {
+        return openShiftAdmin;
+    }
+
+    /**
+     * Delete OpenShift project.
+     */
+    public void delete() {
+        openShift.deleteProject();
+    }
+
+    /**
+     * Process template and create all resources defined there.
+     *
+     * @param templateUrl URL of template to be processed
+     * @param envVariables Map of environment variables to override default values from the template
+     */
+    public void processTemplateAndCreateResources(URL templateUrl, Map<String, String> envVariables) {
+        boolean templateIsFile = templateUrl.getProtocol().equals("file");
+
+        // Used to log into OpenShift
+        OpenShiftBinary oc = openShiftBinaryClient();
+
+        List<String> commandParameters = new ArrayList<>();
+        commandParameters.add(openShiftBinaryPath());
+        commandParameters.add("process");
+        commandParameters.add("-f");
+        commandParameters.add(templateIsFile ? templateUrl.getPath() : templateUrl.toExternalForm());
+        commandParameters.add("--local");
+        commandParameters.add("--ignore-unknown-parameters=true");
+        commandParameters.add("-o");
+        commandParameters.add("yaml");
+        for (Entry<String, String> envVariable : envVariables.entrySet()) {
+            commandParameters.add("-p");
+            commandParameters.add(envVariable.getKey() + "=" + envVariable.getValue());
+        }
+        String completeProcessingCommand = commandParameters.stream().collect(Collectors.joining(" "));
+
+        try (ProcessExecutor executor = new ProcessExecutor()) {
+            File processedTemplate = executor.executeProcessCommandToTempFile(completeProcessingCommand);
+            oc.execute("create", "-n", getName(), "-f", processedTemplate.getAbsolutePath());
+        }
+        // TODO: Temporary workaround to wait until scenario is completely initialized as there is a delay between finishing template creation command
+        // and actual creation of resources on OpenShift. This should be removed when deployments won't be scaled in the beginning and will contain availability check.
+        try {
+            TimeUnit.SECONDS.sleep(10L);
```

Comment:
This has been applied as a workaround, should be refactored.
The point of such construction is that templates can contain placeholders for fields expecting numbers. However Fabric8 client isn't able to process such template - https://github.com/fabric8io/kubernetes-client/issues/959

My suggestion for refactoring - keep executeProcessCommandToTempFile line. Use Fabric8 client to load processedTemplate file into Kubernetes List (or some similar type, please check content of file), this object can be processed by Fabric8. Then create the objects. Then instead of this waiting loop go through the object list loaded by Fabric8 and check that all objects were created using waiter (use just minimal waiting step (100ms ?) as this object creation should be fairly quick).
What do you think about it?

Reply:
Did not know about the loading resource file to Kube list. I will give it a try :)